apps/hash-api/src/graph/knowledge/primitive/entity.ts

```diff
@@ -717,6 +723,74 @@ export const canUpdateEntity: ImpureGraphFunction<
     .canUpdateEntity(actorId, params.entityId)
     .then(({ data }) => data.has_permission);
 
+export const checkPermissionsOnEntity: ImpureGraphFunction<
+  { entity: Pick<Entity, "metadata"> },
+  Promise<UserPermissions>
+> = async (graphContext, { actorId }, params) => {
+  const { entity } = params;
+
+  const { entityId } = entity.metadata.recordId;
+
+  const isAccountGroup =
+    entity.metadata.entityTypeId === SYSTEM_TYPES.entityType.org.schema.$id;
+
+  const isPublicUser = actorId === publicUserAccountId;
```

Comment:
I guess this is to avoid making a dedicated request?

Reply:
While public users can always only view entities, I thought it easier not to bother asking the Graph API anything about them. We can do though if you think safer, it wouldn't be a big deal

Reply:
I think that is reasonable. I don't know how likely it is that this will change at some point, though. We just have to remember that this is encoded here.

Reply:
Okay, cool. I think we'll notice because if we ever try and introduce a feature where public users can edit something, it won't work until we change this.